hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/scm/node/TestQueryNode.java

```diff
@@ -114,25 +117,32 @@ public void testHealthyNodesCount() throws Exception {
   }
 
   @Test
-  @Timeout(10)
   public void testStaleNodesCount() throws Exception {
-    cluster.shutdownHddsDatanode(0);
-    cluster.shutdownHddsDatanode(1);
-
-    GenericTestUtils.waitFor(() ->
-            cluster.getStorageContainerManager().getNodeCount(STALE) == 2,
-        100, 4 * 1000);
-
-    int nodeCount = scmClient.queryNode(null, STALE,
-        HddsProtos.QueryScope.CLUSTER, "").size();
-    assertEquals(2, nodeCount, "Mismatch of expected nodes count");
+    CompletableFuture.runAsync(() -> {
+      cluster.shutdownHddsDatanode(0);
+      cluster.shutdownHddsDatanode(1);
+    }, executor);
```

Comment:
Why does the executor have 2 threads when a single task shuts down both nodes?

Nit: can use `executor` directly, no need for `CompletableFuture`, since we don't use it for checking the results.

```suggestion
    executor.execute(() -> cluster.shutdownHddsDatanode(0));
    executor.execute(() -> cluster.shutdownHddsDatanode(1));
```

Also: `executor` can be local to this test case.

Reply:
> Why does the executor have 2 threads when a single task shuts down both nodes?
> 
> Nit: can use `executor` directly, no need for `CompletableFuture`, since we don't use it for checking the results.
> 
> Also: `executor` can be local to this test case.

Yeah thanks for noticing. I realized that. I have made changes as per your suggestion.